### PR TITLE
Add Expression syntax to doc website repo.

### PR DESCRIPTION
**Pull request by carolxob**
Signed-off-by: carolxob <carolxob@amazon.com>

### Description
Add Expression syntax to doc website repo.

### Issues Resolved
Closes #1267. 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by carolxob** [comment]
@dlvenable This PR is ready for tech review.

File: _data-prepper/expression-syntax.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+<Number | Json Pointer> >= <Number | Json Pointer>
+```
+
+## Examples
+```
+/status_code >= 200 and /status_code < 300
+```
+
+# Equality operators
+Used to test if two value are/are not equivalent.
```
Comment (dlvenable) [inline _data-prepper/expression-syntax.md:67]: Maybe we should reword this.

> Used to test if two value are equivalent or not equivalent.

**Review by dlvenable — APPROVED** [review]
I'm good to merge this, but I also have one minor suggestion for clarity.

File: _data-prepper/expression-syntax.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+# Relational operators
+Tests the relationship of two numeric values. Note, the operands must be a number or Json Pointer that will resolve to a number.
+
+## Syntax
+```
+<Number | Json Pointer> < <Number | Json Pointer>
+<Number | Json Pointer> <= <Number | Json Pointer>
+<Number | Json Pointer> > <Number | Json Pointer>
+<Number | Json Pointer> >= <Number | Json Pointer>
```
Comment (JeffH-AWS) [inline _data-prepper/expression-syntax.md:58]: Capitalize JSON for lines 55-58

File: _data-prepper/expression-syntax.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
+---
+layout: default
+title: Expression syntax
+nav_order: 12
+---
+
+# Supported operators
```
Comment (JeffH-AWS) [inline _data-prepper/expression-syntax.md:7]: I think the title of the page is supposed to match the first heading - but now I'm not 100% sure.

File: _data-prepper/expression-syntax.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+A fundamental value that has no children.
+- Float _(Supports values from 3.40282347 x 10^38 to 1.40239846 x 10^-45)_
+- Integer _(Supports values from -2147483648 to 2147483647)_
+- Boolean _(Supports true or false)_
+- Json Pointer _(See Json Pointer section for details)_
+- String _(Supports Valid Java String characters)_
+- Null _(Supports null check to see if a Json Pointer is present or not)_
+
+## Expression string
+The String that will be parsed for evaluation. Expression String is the highest level of a Data Prepper Expression. Only supports one expression string resulting in a return value. Note, an _Expression String_ is not the same as an _Expression_.
```
Comment (kolchfa-aws) [inline _data-prepper/expression-syntax.md:129]: Should "string" be lowercase here in all instances?

File: _data-prepper/expression-syntax.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+A generic component that contains a _Primary_ or an _Operator_. Expressions may contain expressions. An expressions imminent children can contains 0-1 _Operators_.
+
+## Primary
+
+- _Set_
+- _Priority Expression_
+- _Literal_
+
+## Operator
+Hard coded token that identifies the operation use in an _Expression_.
```
Comment (kolchfa-aws) [inline _data-prepper/expression-syntax.md:144]:
```suggestion
Hard coded token that identifies the operation used in an _Expression_.
```

**Review by kolchfa-aws — APPROVED** [review]
LGTM with minor comments.

**Review by JeffH-AWS — APPROVED** [review]
Looks great! Few minor comments.

File: _data-prepper/expression-syntax.md (hunk 4 above)
Comment (carolxob) [reply _data-prepper/expression-syntax.md:129]: @kolchfa-aws Yes, I think it should.

**Review by natebower — COMMENTED** [review]
@carolxob Please see my comments and changes and let me know if you have any questions. Let me know when you're finished with revisions, and I'll take another look. Thanks!

File: _data-prepper/expression-syntax.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+```
+/status_code == 200 and /message == "Hello world"
+/status_code == 200 or /status_code == 202
+not /status_code in {200, 202}
+/response == null
+/response != null
+```
+
+## Definitions
+
```
[the remark this answers is not in the document]
Comment (carolxob) [reply _data-prepper/expression-syntax.md:122]: @natebower Agree. Also, should we move this section (Definitions) to the beginning or end of the document? It seems out of place where it is currently located in the middle.

[the remark this answers is not in the document]
Comment (natebower) [reply _data-prepper/expression-syntax.md:122]: I would defer to @Naarcha-AWS on this.

File: _data-prepper/expression-syntax.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+- `~0` representing `~`
+- `~1` representing `/`
+
+#### Shorthand syntax (Regex, `\w` = `[A-Za-z_]`)
+```
+/\w+(/\w+)*
+```
+
+##### Shorthand example
```
[the remark this answers is not in the document]
Comment (carolxob) [reply _data-prepper/expression-syntax.md:164]: Yes, this was deliberate, but maybe it should be an H4? It looked like it should be a subset of the previous heading but that may not work.

File: _data-prepper/expression-syntax.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (carolxob) [reply _data-prepper/expression-syntax.md:122]: We are keeping that section where it is for now.

**Pull request merged**